Explorer/Assets/DCL/InWorldCamera/InWorldCamera/UI/InWorldCameraController.cs

```diff
@@ -0,0 +1,155 @@
+        }
+
+        public void Show()
+        {
+            sidebarButton.OnSelect(null);
+            mvcManager.ShowAsync(InWorldCameraController.IssueCommand(new ControllerNoData()));
+
+            bool hasSpace = storageService.StorageStatus.HasFreeSpace;
+            viewInstance!.TakeScreenshotButton.gameObject.SetActive(hasSpace);
+            viewInstance.NoStorageNotification.gameObject.SetActive(!hasSpace);
+        }
+
+        public void Hide(bool isInstant = false)
+        {
+            sidebarButton.OnDeselect(null);
+            viewInstance.HideAsync(default(CancellationToken), isInstant).Forget();
+        }
+
+        protected override UniTask WaitForCloseIntentAsync(CancellationToken ct) =>
+            UniTask.WhenAny(
+                viewInstance!.CloseButton.OnClickAsync(ct),
+                viewInstance.CameraReelButton.OnClickAsync(ct));
+
+        public void PlayScreenshotFX(Texture2D image, float splashDuration, float middlePauseDuration, float transitionDuration)
+        {
+            UIAudioEventsBus.Instance.SendPlayAudioEvent(viewInstance.SFXScreenshotCapture);
+            viewInstance.ScreenshotCaptureAnimation(image, splashDuration, middlePauseDuration, transitionDuration);
+        }
+
+        private void OpenCameraReelGallery()
+        {
+            RequestDisableInWorldCamera();
+
+            mvcManager.ShowAsync(
+                ExplorePanelController.IssueCommand(new ExplorePanelParameter(ExploreSections.CameraReel, BackpackSections.Avatar)));
+        }
+
+        private void RequestTakeScreenshot()
+        {
+            if (!world.Has<TakeScreenshotUIRequest>(camera!.Value))
+                world.Add(camera!.Value, new TakeScreenshotUIRequest { Source = "UI" });
+        }
+
+        private void RequestDisableInWorldCamera()
+        {
+            if (CameraIsActivated())
+                world.Add(camera!.Value, new ToggleInWorldCameraUIRequest { IsEnable = false });
+
+            bool CameraIsActivated() =>
+                !world.Has<ToggleInWorldCameraUIRequest>(camera!.Value) && world.Has<InWorldCamera>(camera!.Value);
+        }
+
+        private void RequestEnableInWorldCamera()
+        {
+            if (CameraIsNotActivated())
+                world.Add(camera!.Value, new ToggleInWorldCameraUIRequest { IsEnable = true });
+
+            bool CameraIsNotActivated() =>
+                !world.Has<ToggleInWorldCameraUIRequest>(camera!.Value) && !world.Has<InWorldCamera>(camera!.Value);
+        }
+
+        private void ToggleShortcutsInfo() =>
+            ToggleShortcutsInfo(!shortcutPanelIsOpen);
+
+        private void ToggleShortcutsInfo(bool toOpen)
+        {
+            if (toOpen)
+            {
+                shortcutsController.LaunchViewLifeCycleAsync(new CanvasOrdering(shortcutsController.Layer, 0), new ControllerNoData(), default(CancellationToken))
+                   .Forget();
+                viewInstance!.ShortcutsInfoButton.OnSelect(null);
```

Comment:
May it happen that the view is not properly initialized yet (previous line)? Could that cause any problem?